cmd/ddev/cmd/composer-create_test.go

```diff
@@ -0,0 +1,87 @@
+package cmd
+
+import (
+	"github.com/stretchr/testify/require"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/ddev/ddev/pkg/config/types"
+	"github.com/ddev/ddev/pkg/ddevapp"
+	"github.com/ddev/ddev/pkg/exec"
+	"github.com/ddev/ddev/pkg/nodeps"
+	"github.com/ddev/ddev/pkg/testcommon"
+	asrt "github.com/stretchr/testify/assert"
+)
+
+func TestComposerCreateCmd(t *testing.T) {
+	// 2022-05-24: I've spent lots of time debugging intermittent `composer create` failures when NFS
+	// is enabled, both on macOS and Windows. As far as I can tell, it only happens in this test, I've
+	// never recreated manually. I do see https://github.com/composer/composer/issues/9627 which seemed
+	// to deal with similar issues in vagrant context, and has a hack now embedded into Composer.
+	if nodeps.PerformanceModeDefault == types.PerformanceModeNFS {
+		t.Skip("Composer has strange behavior in NFS context, so skipping")
+	}
+	assert := asrt.New(t)
+
+	origDir, err := os.Getwd()
+	assert.NoError(err)
+
+	types := ddevapp.GetValidAppTypes()
+	if os.Getenv("GOTEST_SHORT") != "" {
+		types = []string{nodeps.AppTypePHP, nodeps.AppTypeDrupal10}
+	}
```

Comment:
@rfay I continued to work on this and added some tests, while I still would appreciate you trying this out and vouching on the approach (🤞) I have an immediate question here..  I added this by looking and some other tests but the tests suite here on GH actions run it with GOTEST_SHORT set to 12 which not exactly sure what that mean. I think it's sensible for this test to run for every single project, is there a proper way of doing that.. shall I remove this if clause and always make it full, or maybe do a `!= 'false'` so it can be set locally... thanks!

Reply:
I guess you mean you want it to run for every project type. 

This test is extremely resource-intensive, so that's why it's only run with the one. 

GOTEST_SHORT=12 means that by default TestDdevFullSiteSetup and another test or two will only use stanza 12 in the test setup, 12 is drupal10, https://github.com/ddev/ddev/blob/64c9e469427785debfdda7db2e8caf1b8632433c/pkg/ddevapp/ddevapp_test.go#L225-L238

If GOTEST_SHORT is empty, it allows running all. In this case it's just doing php and drupal10 when GOTEST_SHORT is set to *anything*.

Does that answer your question?

Reply:
Not really, but I understood of what GOTEST_SHORT does with an integer. My question is, can I allow my test of composer create to run for every project type?

Reply:
We also sorted this one out on our zoom call